File: Backend/Models/Subject.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models;


public class Subject
{
    public Guid SubjectId{ get; set; }
    [Required]
    public string Name { get; set; } = string.Empty;
    [Required]
    [Range(1, 20)]
    public float Credits { get; set; }
    [Required]
    public string Icon { get; set; } = string.Empty;
    [Required]
    public SubjectColor Color { get; set; }
}

File: frontend/src/types/Subject.ts
export interface Subject {
    subjectId: string
    name: string
    credits: number
    icon: string
    color: string
}

export interface SubjectWithStats extends Subject {
    assignmentsCount: number
    assignmentsPending: number
    assignmentsInProgress: number
    assignmentsReview: number
    assignmentsCompleted: number
    examsCount: number
    examsAverageGrade?: number
}